Tasks/XamarinAndroid/xamarinandroid.ts

```diff
@@ -1,151 +1,99 @@
-
-// find jdk to be used during the build
-var jdkSelection = tl.getInput('jdkSelection');
-if(!jdkSelection) {
-    jdkSelection = 'JDKVersion'; //fallback to JDKVersion for older version of tasks
-}
-var specifiedJavaHome = null;
-
-if (jdkSelection == 'JDKVersion') {
-    tl.debug('Using JDK version to find JDK path');
-    var jdkVersion = tl.getInput('jdkVersion');
-    var jdkArchitecture = tl.getInput('jdkArchitecture');
-
-    if(jdkVersion != 'default') {
-        // jdkVersion should be in the form of 1.7, 1.8, or 1.10
-        // jdkArchitecture is either x64 or x86
-        // envName for version 1.7 and x64 would be "JAVA_HOME_7_X64"
-        var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
-        specifiedJavaHome = tl.getVariable(envName);
-        if (!specifiedJavaHome) {
-            tl.error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
-            tl.exit(1);
-        }
-    }
-}
-else {
-    tl.debug('Using path from user input to find JDK');
-    var jdkUserInputPath = tl.getPathInput('jdkUserInputPath', true, true);
-    specifiedJavaHome = jdkUserInputPath;
-}
-
-//find xbuild location to use
-var xbuildToolPath = tl.which('xbuild');
-if(xbuildLocation) {
-    xbuildToolPath = path.join(xbuildLocation, 'xbuild');
-    if (!tl.exist(xbuildToolPath)) {
-        xbuildToolPath = path.join(xbuildLocation, 'xbuild.exe');
-    }
-    tl.checkPath(xbuildToolPath, 'xbuild');
-}
-if(!xbuildToolPath) {
-    tl.error('xbuild was not found in the path.');
-    tl.exit(1);
-}
-
-var runxbuild = function (fn) {
-    return Q.fcall( () => {
-        var xbuild = tl.createToolRunner(xbuildToolPath);
-        xbuild.pathArg(fn);
-
-		if(clean) {
-            xbuild.arg('/t:Clean');
+import { ToolRunner } from 'vsts-task-lib/toolrunner';
+
+async function run() {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+        //read inputs
+        let project: string = tl.getPathInput('project', true);
+        let target: string = tl.getInput('target');
+        let outputDir: string = tl.getInput('outputDir');
+        let configuration: string = tl.getInput('configuration');
+        let createAppPackage: boolean = tl.getBoolInput('createAppPackage');
+        let clean: boolean = tl.getBoolInput('clean');
+        let xbuildLocation: string = tl.getPathInput('msbuildLocation');
+        let msbuildArguments: string = tl.getInput('msbuildArguments');
+
+        // find jdk to be used during the build
+        let jdkSelection: string = tl.getInput('jdkSelection');
+        if (!jdkSelection) {
+            jdkSelection = 'JDKVersion'; //fallback to JDKVersion for older version of tasks
         }
-        if (target) {
-            xbuild.arg('/t:' + target);
+        let specifiedJavaHome = null;
+
+        if (jdkSelection == 'JDKVersion') {
```

Comment:
We now have a Java-Common module for finding the JDK (this is taken from Maven task):
`specifiedJavaHome = javacommons.findJavaHome(jdkVersion, jdkArchitecture);`

I guess originally this ts file only runs on Linux/Mac as this doesn't seem to work on hosted Windows build?